pelican/log.py

```diff
@@ -100,6 +101,11 @@ def filter(self, record):
         if record.levelno > logging.WARN:
             return True
 
+        # filter record if user ignored it by id
+        id = record.__dict__.get('id', None)
+        if id in self._ignore_ids:
+            return False
+
```

Comment:
If you want to do a variable assignment here, it might be better to use `_id` as a name to avoid shadowing Python's `id` built-in. Otherwise, you could of course just do this (the second argument to `.get()` defaults to `None` if one isn't explicitly given):

```python
if record.__dict__.get('id') in self._ignore_ids:
    return False
```